pootle/apps/pootle_statistics/proxy.py

```diff
@@ -0,0 +1,236 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) Pootle contributors.
+#
+# This file is a part of the Pootle project. It is distributed under the GPL3
+# or later license. See the LICENSE file for a copy of the license and the
+# AUTHORS file for copyright and authorship information.
+
+from django.core.urlresolvers import reverse
+from django.template.defaultfilters import truncatechars
+from django.utils.functional import cached_property
+
+from accounts.proxy import DisplayUser
+from pootle.core.primitives import PrefixedDict
+from pootle.core.url_helpers import get_editor_filter, split_pootle_path
+from pootle.core.utils import dateformat
+from pootle_misc.checks import check_names
+from pootle_store.constants import FUZZY, TRANSLATED
+from pootle_store.fields import to_python
+
+from .models import SubmissionFields, SubmissionTypes, TranslationActionTypes
+
+
+class SubmissionProxy(object):
+    """Wraps a dictionary of submission values, which is useful for wrapping
+    results from qs.values calls
+    """
+
+    fields = (
+        "type",
+        "old_value",
+        "new_value",
+        "creation_time",
+        "field")
+    qc_fields = (
+        "quality_check_id",
+        "quality_check__name")
+    submitter_fields = (
+        "submitter_id",
+        "submitter__username",
+        "submitter__email",
+        "submitter__full_name")
+    suggestion_fields = (
+        "suggestion_id",
+        "suggestion__target_f", )
+    suggestion_reviewer_fields = (
+        "suggestion__reviewer__full_name",
+        "suggestion__reviewer__email",
+        "suggestion__reviewer__username")
+    suggestion_user_fields = (
+        "suggestion__user__full_name",
+        "suggestion__user__email",
```

Comment:
theres quite a lot in the new class - it deals with 2 different situations - timeline and get_submission_info